x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/support/SecurityIndexManager.java

```diff
@@ -167,7 +161,8 @@ public void clusterChanged(ClusterChangedEvent event) {
         final Version mappingVersion = oldestIndexMappingVersion(event.state());
         final ClusterHealthStatus indexStatus = indexMetaData == null ? null :
             new ClusterIndexHealth(indexMetaData, event.state().getRoutingTable().index(indexMetaData.getIndex())).getStatus();
-        final State newState = new State(indexExists, isIndexUpToDate, indexAvailable, mappingIsUpToDate, mappingVersion, indexStatus);
+        final State newState = new State(indexExists, isIndexUpToDate, indexAvailable, mappingIsUpToDate, mappingVersion,
+            indexMetaData == null ? INTERNAL_SECURITY_INDEX : indexMetaData.getIndex().getName(), indexStatus);
```

Comment:
Break `indexMetaData == null ? INTERNAL_SECURITY_INDEX : indexMetaData.getIndex().getName()` out in a variable for readability's sake?